src/providers/GlobalContext/GlobalContext.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useReducer, createContext } from 'react';
+
+const GlobalContext = createContext();
+
+export function GlobalContextProvider({ children }) {
+  const searchTextReducer = (state, action) => {
+    switch (action.type) {
+      case 'set':
+        return action.payload;
+      default:
+        return state;
+    }
+  };
+
+  // const lightTheme = {
+  //   bg: 'white',
+  //   text: 'black',
+  // };
+
+  // const darkTheme = {
+  //   bg: '#303030',
+  //   text: 'white',
+  // };
+
+  const themeReducer = (state, action) => {
+    switch (action.type) {
+      case 'setLightTheme':
+        return 'lightTheme';
+      case 'setDarkTheme':
+        return 'darkTheme';
+      default:
+        return state;
+    }
+  };
+
+  const [searchState, searchTextDispatch] = useReducer(searchTextReducer, 'wizeline');
+  const [themeState, themeDispatch] = useReducer(themeReducer, 'lightTheme');
+
```

Comment:
Once you merge your reducers you will end up with a single state and a single dispatch function. Try testing the reducer actions to make sure your actions mutate the state properly.